### ISSUE TITLE
Search path issues when subsetting RLum.Analysis objects

### ISSUE THREAD

**Issue by tzerk** [R-Lum/Luminescence#53]
For some reason when a call to `subset.RLum.Analysis()` is surrounded by a function `get_RLum.Analysis()` searches `x` (see example below) in the global environment rather than in the environment of the calling function. This, of course, results in an error as `x` is not found within the global environment; this is unless we assign it before calling `foo()`.

Minimal reproducible example:
```r
library(Luminescence)

# An RLum.Analysis object
data("ExampleData.RLum.Analysis")
rlum <- IRSAR.RF.Data

# Enclosing function to subset()
foo <- function(obj) {
  x <- "RF"
  rlum_sub <- subset(obj, recordType == x)
}

# Fails (enclosing function):
foo(rlum)

# Works (interactive):
subset(rlum, recordType == "RF")

# Works (define 'x' in the global environment before calling foo())
x <- "RF"
foo(rlum)
```

**Comment by tzerk** [comment]
After studying the ['Environments' section of Wickhams 'Advanced R'](http://adv-r.had.co.nz/Environments.html) I found that this issue was within our call to `eval()` we use to in `get_RLum()` to subset the object. More specifically, we used `parent.frame(1)` as the enclosing environment for R to find the names associated with the subsetting. Albeit being the "wrong" environment, this still worked from within the interactive environment, as lexical scoping let R find the names in `.GlobalEnv`.

Now, if either `subset.RLum.Analysis()` or `get_RLum()` was called from within a function, in which also the variables for subsetting were defined, R couldn't find the names in neither `parent.frame(1)` or in the `.GlobalEnv`.  

With the new argument `env` in `get_RLum()` we should now be able to pass the correct enclosing environment to `eval()`.
